File: aufgabe01/nmeyering/FizzBuzz.hs
-- ausführen als `runhaskell FizzBuzz.hs` pls, kthxbai
data FizzBuzz = FizzBuzz Integer

instance Show FizzBuzz where
  show (FizzBuzz n)
    | mod n 3 == 0 && mod n 5 == 0 = "FizzBuzz"
    | mod n 3 == 0 = "Fizz"
    | mod n 5 == 0 = "Buzz"
    | otherwise = show n

main = do
  line <- getLine
  print $ FizzBuzz (read line :: Integer)

File: FizzBuzz.java

public class FizzBuzz { 
    public static void main(String[] args) { 
	int a = Integer.parseInt(args[0]);
	for (int i = 1; i <= a; i++) 
	{
		if ((i % 3 == 0) && (i % 5 == 0))
			System.out.println("Fizzbuzz");
		else if (i % 3 == 0)
			System.out.println("Fizz");
		else if (i % 5 == 0)
			System.out.println("Buzz");
		else System.out.println(""+i);
	}
   }



} 
